### ISSUE TITLE
fireEvent.press does not work on TouchableOpacity

### ISSUE THREAD

**Issue by hsjoberg** [testing-library/native-testing-library#113]
- `react-native` or `expo`: react-native
- `native-testing-library` version: 5.0.3
- `jest-preset`: @testing-library/react-native
- `react-native`: 0.62.0-rc.3
- `node` version: v12.12.0

### Relevant code or config:

```jsx
import React from "react";
import { TouchableOpacity } from "react-native";
import { render, fireEvent } from "@testing-library/react-native";

it("test", async () => {
  const { findByTestId } = render(
    <TouchableOpacity testID="button" onPress={() => console.log("TEST")} />
  );
  const button = await findByTestId("button");
  fireEvent.press(button);
});
```

### What you did:

In  _0.62.0-rc.3_, `fireEvent.press` does not seem to work at all on `TouchableOpacity`. Trying on for example a `Button` seems to work fine, just not `TouchableOpacity` (and presumably similar components).

### Reproduction:

See example code above. Running it you should see a console logged "TEST" message, in 0.62.0-rc.3, it does not appear.

### Can you help us fix this issue by submitting a pull request?

Yes, though I don't know what's wrong yet.

**Comment by akiver** [comment]
I had the same issue after upgrading to RN 0.62.
I think it's related to this issue https://github.com/facebook/react-native/issues/27721.

`displayName` is missing and native-testing-library rely on it to fire events.
 
```
[Function: Component] {
      displayName: '', <- Supposed to be TouchableOpacity or TouchableHighlight
      '$$typeof': Symbol(react.forward_ref),
      render: [Function]
 }
```

**Comment by halilb** [comment]
Our tests have also broken after upgrading to RN 0.62. I've applied a temporary fix to make the tests pass again by mocking `TouchableOpacity` with `TouchableHighlight` in `jest.setup.js` file:

```
jest.mock(
  'react-native/Libraries/Components/Touchable/TouchableOpacity.js',
  () => {
    const { TouchableHighlight } = require('react-native')
    const MockTouchable = props => {
      return <TouchableHighlight {...props} />
    }
    return MockTouchable
  }
)
```

**Comment by hyochan** [comment]
Thanks for the issue. Having same problem. When I've updated the snapshot I see all the `TouchableOpacity` are gone.
![image](https://user-images.githubusercontent.com/27461460/78248644-03bbc380-7528-11ea-9688-4d318865a149.png)

**Comment by mym0404** [comment]
I am experiencing exactly the same thing.

**Comment by andrico1234** [comment]
@halilb Your fix worked for me!

**Comment by kevinpiac** [comment]
> jest.mock(
>   'react-native/Libraries/Components/Touchable/TouchableOpacity.js',
>   () => {
>     const { TouchableHighlight } = require('react-native')
>     const MockTouchable = props => {
>       return <TouchableHighlight {...props} />
>     }
>     MockTouchable.displayName = 'TouchableOpacity'
> 
>     return MockTouchable
>   }
> )


@halilb Thanks for this solution, however, this fix is not totally working for me since it breaks a lot of other unit tests. 🤕 

However, this simple snippet as mentioned [here ](https://github.com/facebook/react-native/issues/27721#issuecomment-609052183) works fine for me :)

I hope it helps.

> jest.mock('react-native/Libraries/Components/Touchable/TouchableOpacity', () => 'TouchableOpacity');

**Comment by Zloka** [comment]
I'm running into the same issue when trying to the `Button` component of `react-native-paper`. It seems to use TouchableWithoutFeedback under the hood.

Any pointers on how I'd go about mocking that out? Using the above (but for TouchableWithoutFeedback) doesn't seem to do the trick 🤔

**Comment by michelalbers** [comment]
Can confirm the issue.

**Comment by daveols** [comment]
I've fixed this in #136 🎉

**Comment by daveols** [comment]
You can now upgrade to version `6.0.0` for this fix! 🚀
